specs/ThrottlingControlScript.md

```diff
@@ -0,0 +1,325 @@
+  [propput] HRESULT ThrottlingIntervalPreferenceForeground([in] UINT32 value);
+
+  /// The preferred wake up interval (in milliseconds) to use for throttleable
+  /// JavaScript tasks (`setTimeout` and `setInterval`), when the WebView is in
+  /// background state, with no intensive throttling. A WebView is in background
+  /// state when its `IsVisible` property is `FALSE`. Intensive throttling is a
+  /// substate of background state. For more details about intensive throttling,
+  /// see [Intensive throttling of Javascript timer wake ups](https://chromestatus.com/feature/4718288976216064).
+  ///
+  /// A wake up interval is the amount of time that needs to pass before the
+  /// WebView2 Runtime checks for new timer tasks to run.
+  ///
+  /// The WebView2 Runtime will try to respect the preferred interval set by the
+  /// application, but the effective value will be constrained by resource and
+  /// platform limitations. Setting a value of `0` means a preference of no
+  /// throttling to be applied. The default value is a constant determined by
+  /// the running version of the WebView2 Runtime. All other background state
+  /// policies (including intensive throttling) are effective independently of
+  /// this setting.
+  ///
+  /// For example, an application might use a background value of 100 ms to
+  /// relax the default background value (usually 1000 ms).
+  [propget] HRESULT ThrottlingIntervalPreferenceBackground([out, retval] UINT32* value);
+  /// Sets the `ThrottlingIntervalPreferenceBackground` property.
+  [propput] HRESULT ThrottlingIntervalPreferenceBackground([in] UINT32 value);
+
+  /// The preferred wake up interval (in milliseconds) to use for throttleable
+  /// JavaScript tasks (`setTimeout` and `setInterval`), when the WebView is in
+  /// background state with intensive throttling. Intensive throttling is a
+  /// substate of background state. For more details about intensive
+  /// throttling, see
+  /// [Intensive throttling of Javascript timer wake ups](https://chromestatus.com/feature/4718288976216064).
+  ///
+  /// A wake up interval is the amount of time that needs to pass before the
+  /// WebView2 Runtime checks for new timer tasks to run.
+  ///
+  /// The WebView2 Runtime will try to respect the preferred interval set by the
+  /// application, but the effective value will be constrained by resource and
+  /// platform limitations. Setting a value of `0` means a preference of no
+  /// throttling to be applied. The default value is a constant determined by
+  /// the running version of the WebView2 Runtime.
+  [propget] HRESULT ThrottlingIntervalPreferenceIntensive([out, retval] UINT32* value);
+  /// Sets the `ThrottlingIntervalPreferenceIntensive` property.
+  [propput] HRESULT ThrottlingIntervalPreferenceIntensive([in] UINT32 value);
+
+  /// The preferred wake up interval (in milliseconds) to use for throttleable
+  /// JavaScript tasks (`setTimeout` and `setInterval`), in frames whose
```

Comment:
This Interval Preference is unique among the four, in that it applies to frame hosted content instead of this WebView's direct content.  Maybe worth differentiating e.g. `ThrottlingIntervalPreferenceIsolatedFrames`

Reply:
**For review discussion**
I agree and intent to update the name unless there's a different outcome during review.

Reply:
See above.